dev-academy/dao-governance/02-dao-contract.md

```diff
@@ -4,68 +4,97 @@ sidebar_position: 2
-parameters.
+```shell
+npx @cosmjs/cli@^0.28.1 --init https://raw.githubusercontent.com/InterWasm/cw-plus-helpers/main/base.ts --init https://raw.githubusercontent.com/InterWasm/cw-plus-helpers/main/cw3-flex-multisig.ts
+```
+```typescript
+# Tab 1
+const [addr, client] = await useOptions(malagaOptions).setup('password', '.account1.key');
+console.log(addr);
+```
+```typescript
+# Tab 2
+const [addr, client] = await useOptions(malagaOptions).setup('password', '.account2.key');
+console.log(addr);
+```
+```typescript
+# Tab 3
+const [addr, client] = await useOptions(malagaOptions).setup('password', '.account3.key');
+console.log(addr);
+```
+```typescript
+# Tab 4
+const [addr, client] = await useOptions(malagaOptions).setup('password', '.account4.key');
+console.log(addr);
+```
+Copy and note the wallet addresses down.
+
+## Architecture
 
+We will use `cw4-group` and `cw3-flex-multisig` contracts for the DAO. The advantage of having two separate governance contracts is the provided flexibility. The `cw4-group` contract contains member details, while the `cw3-flex-multisig` contract contains the cw4-group address, the voting threshold and the voting period. With this architecture, one group can have multiple multisig accounts with different voting parameters.
 ## cw4-group Setup
 
-On the first terminal, let us start setting up a wallet and launch the DAO.
+On the first Terminal window, let us set up a wallet address to act as the group administrator and launch the DAO with the 4 member addresses generated earlier.
 
-Launch CosmJS client:
+Initialize the CosmJS CLI session:
 
 ```shell
-npx @cosmjs/cli@^0.26 --init https://raw.githubusercontent.com/InterWasm/cw-plus-helpers/main/base.ts --init https://raw.githubusercontent.com/InterWasm/cw-plus-helpers/main/cw4-group.ts
+npx @cosmjs/cli@^0.28.1 --init https://raw.githubusercontent.com/InterWasm/cw-plus-helpers/main/base.ts --init https://raw.githubusercontent.com/InterWasm/cw-plus-helpers/main/cw4-group.ts
 ```
 
-Now store and instantiate the contract:
+Now set up a wallet address before deploying and instantiating the cw4-group contract:
 
 ```typescript
-const [addr, client] = await useOptions(cliffnetOptions).setup('password', '.main.key');
-const contract = CW4Group(client, cliffnetOptions.fees);
-const codeId = await contract.upload(addr);
+const [addr, client] = await useOptions(malagaOptions).setup('password', '.main.key');
+const contract = CW4Group(client, malagaOptions);
+const codeId = await contract.upload(addr, malagaOptions);
+console.log(codeId);
+
+//Preparing the instantiation message, make sure to 
+//replace the member addresses with the ones generated earlier for each member
+.editor
 
-// .editor
 const initMsg = {
     admin: addr,
     members: [
         {
-            addr: "wasm19pxmaaq8v6zgs5umtz8eqtpwcfgdj2nlhqhff3",
+            addr: "Member Address #1",
```

Comment:
We need to put down legit malaga addresses to make the script work

Reply:
The user is directed to generate the member addresses during environment setup and edit the instantiation message accordingly (to be able to vote with those keys later on). Could pass the member addresses in as variables, alternatively.